test/Conversion/ArcToLLVM/lower-arc-to-llvm.mlir

```diff
@@ -222,3 +222,56 @@ func.func @WriteArray(%arg0: !arc.state<!hw.array<4xi1>>, %arg1: !hw.array<4xi1>
   arc.state_write %arg0 = %arg1 : <!hw.array<4xi1>>
   return
 }
+
+func.func @assert_with_cond() {
+  // CHECK-LABEL: @assert_with_cond(
+  // CHECK-NEXT:    [[COND:%.+]] = llvm.mlir.constant(false) : i1
+  // CHECK-NEXT:    llvm.cond_br [[COND]], ^[[TRUE_BRANCH:[[:alnum:]]+]], ^[[FALSE_BRANCH:[[:alnum:]]+]]
+  // CHECK-NEXT:    ^[[TRUE_BRANCH]]:
+  // CHECK-NEXT:      llvm.br ^[[END:[[:alnum:]]+]]
+  // CHECK-NEXT:    ^[[FALSE_BRANCH]]:
+  // CHECK-NEXT:      llvm.call @abort() : () -> ()
+  // CHECK-NEXT:      llvm.br ^[[END:[[:alnum:]]+]]
+  // CHECK-NEXT:    ^[[END]]:
+  // CHECK-NEXT:      llvm.return
+  // CHECK-NEXT:  }
+  %cond = hw.constant 0 : i1
+  arc.assert %cond
+  return
+}
+
+func.func @assert_with_cond_clock() {
+  // CHECK-LABEL: @assert_with_cond_clock(
+  // CHECK-NEXT:    [[COND:%.+]]  = llvm.mlir.constant(false) : i1
+  // CHECK-NEXT:    [[CLOCK:%.+]] = llvm.mlir.constant(true) : i1
+  // CHECK-NEXT:    llvm.cond_br [[COND]], ^[[TRUE_BRANCH:[[:alnum:]]+]], ^[[FALSE_BRANCH:[[:alnum:]]+]]
+  // CHECK-NEXT:    ^[[TRUE_BRANCH]]:
+  // CHECK-NEXT:      llvm.br ^[[END:[[:alnum:]]+]]
+  // CHECK-NEXT:    ^[[FALSE_BRANCH]]:
+  // CHECK-NEXT:      llvm.call @abort() : () -> ()
+  // CHECK-NEXT:      llvm.br ^[[END:[[:alnum:]]+]]
+  // CHECK-NEXT:    ^[[END]]:
+  // CHECK-NEXT:      llvm.return
+  // CHECK-NEXT:  }
+  %cond = hw.constant 0 : i1
+  %clock = hw.constant 1: i1
+  arc.assert %cond, %clock
```

Comment:
I think at this stage of the Arc lowering pipeline, the `clock` operand should not be present anymore. Either it is folded into the `cond` operand or it is gone because the assert is in a function that is called only at the posedge of this clock. So the lowering pass to LLVM would just error out. But the clock operand will be useful in the lowering from `verif` and be dealt with in the passes between those two lowerings.
Also, shouldn't the clock operand be of `seq.clock` type?
@fabianschuiki What do you think?

Reply:
I put a clock operand so we can make use of it for further steps, I am not sure why we need it in `assert` which checks the condition.

Reply:
Yes, having the optional clock operand makes sense. But the clock operand just provides the raw value of the clock signal, but the assert is typically only checked at one or both of the clock edges (see `ltl.clock` operation). This behavior needs to be lowered and will probably happen before the ArcToLLVM pass. During this process this edge-checking logic could be folded into the condition.

Reply:
>the clock operand should not be present anymore. Either it is folded into the cond operand or it is gone because the assert is in a function that is called only at the posedge of this clock.

@maerhart, can you explain this a little further?